diff --git a/core/modules/filter/filter.admin.es6.js b/core/modules/filter/filter.admin.es6.js
index 731ed5b7a4..6e83c5df6f 100644
--- a/core/modules/filter/filter.admin.es6.js
+++ b/core/modules/filter/filter.admin.es6.js
@@ -25,6 +25,10 @@
         const $row = $context
           .find(`#${$checkbox.attr('id').replace(/-status$/, '-weight')}`)
           .closest('tr');
+
+        const $table = $row.parents('table');
+        const $tableLabels = $table.siblings();
+
         // Retrieve the vertical tab belonging to this filter.
         const $filterSettings = $context.find(
           `[data-drupal-selector='${$checkbox
@@ -44,6 +48,9 @@
               // On very narrow viewports, Vertical Tabs are disabled.
               $filterSettings.show();
             }
+
+            // Show the table labels
+            $tableLabels.show();
           } else {
             $row.hide();
             if (filterSettingsTab) {
@@ -52,6 +59,18 @@
               // On very narrow viewports, Vertical Tabs are disabled.
               $filterSettings.hide();
             }
+
+            // Hide the table labels if there are no visible rows.
+            let visibleRows = false;
+            $table.find('tr').each(function() {
+              if ($(this).is(':visible')) {
+                visibleRows = true;
+                return;
+              }
+            });
+            if (!visibleRows) {
+              $tableLabels.hide();
+            }
           }
           // Restripe table after toggling visibility of table row.
           Drupal.tableDrag['filter-order'].restripeTable();
diff --git a/core/modules/filter/filter.admin.js b/core/modules/filter/filter.admin.js
index ea43f14343..48ba824a9d 100644
--- a/core/modules/filter/filter.admin.js
+++ b/core/modules/filter/filter.admin.js
@@ -7,12 +7,14 @@
 (function ($, Drupal) {
   Drupal.behaviors.filterStatus = {
     attach: function attach(context, settings) {
-      var $context = $(context);
+      const $context = $(context);
       once('filter-status', '#filters-status-wrapper input.form-checkbox', context).forEach(function (checkbox) {
-        var $checkbox = $(checkbox);
-        var $row = $context.find("#".concat($checkbox.attr('id').replace(/-status$/, '-weight'))).closest('tr');
-        var $filterSettings = $context.find("[data-drupal-selector='".concat($checkbox.attr('id').replace(/-status$/, '-settings'), "']"));
-        var filterSettingsTab = $filterSettings.data('verticalTab');
+        const $checkbox = $(checkbox);
+        const $row = $context.find("#".concat($checkbox.attr('id').replace(/-status$/, '-weight'))).closest('tr');
+        const $table = $row.parents('table');
+        const $tableLabels = $table.siblings();
+        const $filterSettings = $context.find("[data-drupal-selector='".concat($checkbox.attr('id').replace(/-status$/, '-settings'), "']"));
+        const filterSettingsTab = $filterSettings.data('verticalTab');
         $checkbox.on('click.filterUpdate', function () {
           if ($checkbox.is(':checked')) {
             $row.show();
@@ -21,6 +23,7 @@
             } else {
               $filterSettings.show();
             }
+            $tableLabels.show();
           } else {
             $row.hide();
             if (filterSettingsTab) {
@@ -28,6 +31,16 @@
             } else {
               $filterSettings.hide();
             }
+            let visibleRows = false;
+            $table.find('tr').each(function () {
+              if ($(this).is(':visible')) {
+                visibleRows = true;
+                return;
+              }
+            });
+            if (!visibleRows) {
+              $tableLabels.hide();
+            }
           }
           Drupal.tableDrag['filter-order'].restripeTable();
         });
